Source/Core/DolphinQt/Config/Mapping/IOWindow.cpp

```diff
@@ -532,7 +562,29 @@ void IOWindow::UpdateDeviceList()
       QString::fromStdString(m_controller->GetDefaultDevice().ToString()));
 }
 
-InputStateDelegate::InputStateDelegate(IOWindow* parent) : QItemDelegate(parent), m_parent(parent)
+void IOWindow::UpdateExpression(std::string new_expression)
+{
+  const auto lock = m_controller->GetStateLock();
+  if (new_expression == m_reference->GetExpression())
+    return;
+
+  const auto error = m_reference->SetExpression(std::move(new_expression));
+  const auto status = m_reference->GetParseStatus();
+  m_controller->UpdateSingleControlReference(g_controller_interface, m_reference);
+
+  if (error)
+    m_parse_text->setText(QString::fromStdString(*error));
+  else if (status == ciface::ExpressionParser::ParseStatus::EmptyExpression)
+    m_parse_text->setText(QString());
+  else if (status != ciface::ExpressionParser::ParseStatus::Successful)
+    m_parse_text->setText(tr("Invalid Expression."));
+  else
+    m_parse_text->setText(tr("Success."));
```

Comment:
Maybe if the parse is successful the result text can be hidden and the new `m_result_value` can take up the whole row?
Similarly, if the parse fails you could hide `m_result_value`?